src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/GraphSymbolUtils.cs

```diff
@@ -143,18 +143,161 @@ private static IEnumerable<INamedTypeSymbol> GetAllViewTypesFromPXGraphOrPXGraph
+            {
+                IEnumerable<(MethodDeclarationSyntax, IMethodSymbol)> dels = graph.GetViewDelegatesFromGraph(views, pxContext, cancellation);
+
+                foreach ((MethodDeclarationSyntax, IMethodSymbol) d in dels)
+                {
+                    delegates.Add(d.Item2.Name, d);
+                }
+            }
+
+            void AddDelegatesFromGraphExtension(Dictionary<string, (MethodDeclarationSyntax, IMethodSymbol)> delegates, ITypeSymbol graphExt)
+            {
+                IEnumerable<(MethodDeclarationSyntax, IMethodSymbol)> dels = graphExt.GetViewDelegatesFromGraphOrGraphExtension(views, pxContext, cancellation);
+
+                foreach ((MethodDeclarationSyntax, IMethodSymbol) d in dels)
+                {
+                    delegates[d.Item2.Name] = d;
+                }
+            }
+        }
+
+        private static IEnumerable<T> GetViewInfoFromGraphExtension<T>(ITypeSymbol graphExtension, PXContext pxContext,
+            Action<Dictionary<string, T>, ITypeSymbol> addGraphViewInfo,
+            Action<Dictionary<string, T>, ITypeSymbol> addGraphExtensionViewInfo)
+        {
+            IEnumerable<T> empty = Enumerable.Empty<T>();
+
+            if (!graphExtension.InheritsFrom(pxContext.PXGraphExtensionType) || !graphExtension.BaseType.IsGenericType)
+            {
+                return empty;
+            }
+
+            INamedTypeSymbol baseType = graphExtension.BaseType;
+            Dictionary<string, T> infoByView = new Dictionary<string, T>(StringComparer.OrdinalIgnoreCase);
+
+            for (int i = baseType.TypeArguments.Length - 1; i >= 0; i--)
+            {
+                ITypeSymbol argType = baseType.TypeArguments[i];
+
+                if (i == baseType.TypeArguments.Length - 1)
+                {
+                    if (!argType.IsPXGraph(pxContext))
+                    {
+                        return empty;
+                    }
+
+                    addGraphViewInfo(infoByView, argType);
+                }
+                else
+                {
+                    if (!argType.IsPXGraphExtension(pxContext))
+                    {
+                        return empty;
+                    }
+
+                    addGraphExtensionViewInfo(infoByView, argType);
+                }
+            }
+
+            addGraphExtensionViewInfo(infoByView, graphExtension);
+
+            return infoByView.Values;
+        }
+
+        private static IEnumerable<(MethodDeclarationSyntax Node, IMethodSymbol Symbol)> GetViewDelegatesFromGraphOrGraphExtension(
+            this ITypeSymbol graphOrExtension, IEnumerable<ISymbol> views, PXContext pxContext, CancellationToken cancellation)
+        {
+            foreach (ISymbol member in graphOrExtension.GetMembers())
+            {
+                cancellation.ThrowIfCancellationRequested();
+
+                if (!(member is IMethodSymbol method))
```

Comment:
@metallmonster @SENya1990 in a recent pull request I've added a helper method `GetMethods` which does exactly what is needed here

Reply:
@dropsonic yes, I've seen it. But it seems like he will always create an immutable array which is less perfomant. Maybe, because GetMembers already returns Immutable array that helper could return `IEnumerable<T>` since inside already underlies an ImmutableArray which can't be changed?

Reply:
@SENya1990 all Roslyn methods return immutable collections, so I don't think we should consider it as "less performant" and worry about it at all.

Reply:
ok